src/rules/effects/prefer-concat-latest-from.ts

```diff
@@ -21,43 +30,96 @@ export default createRule<Options, MessageIds>({
     version: '>=12.0.0',
     docs: {
       category: 'Possible Errors',
-      description:
-        'Use `concatLatestFrom` instead of `withLatestFrom` to prevent the selector from firing until the correct `Action` is dispatched.',
+      description: `Use \`${concatLatestFromKeyword}\` instead of \`${withLatestFromKeyword}\` to prevent the selector from firing until the correct \`Action\` is dispatched.`,
       recommended: 'warn',
-      suggestion: true,
     },
-    schema: [],
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          strict: {
+            type: 'boolean',
+            default: defaultOptions.strict,
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
     messages: {
-      [messageId]: 'Use `concatLatestFrom` instead of `withLatestFrom`.',
-      [messageIdSuggest]: 'Replace `withLatestFrom` with `concatLatestFrom`.',
+      [messageId]: `Use \`${concatLatestFromKeyword}\` instead of \`${withLatestFromKeyword}\`.`,
     },
   },
-  defaultOptions: [],
-  create: (context) => {
+  defaultOptions: [defaultOptions],
+  create: (context, [options]) => {
+    const sourceCode = context.getSourceCode()
+    const selector = getSelector(context, options)
+
+    if (!selector) {
+      return {}
+    }
+
     return {
-      [`${createEffectExpression} > [body] > [arguments] > CallExpression > Identifier[name='withLatestFrom']`](
-        node: TSESTree.Identifier,
-      ) {
+      [selector](node: WithLatestFromIdentifier) {
         context.report({
           node,
           messageId,
-          suggest: [
-            {
-              messageId: messageIdSuggest,
-              fix: (fixer) => {
-                return [fixer.replaceText(node, 'concatLatestFrom')].concat(
-                  getImportAddFix({
-                    fixer,
-                    importName: 'concatLatestFrom',
-                    moduleName: NGRX_MODULE_PATHS.effects,
-                    node,
-                  }),
-                )
-              },
-            },
-          ],
+          fix: (fixer) => getFixes(sourceCode, fixer, node),
         })
       },
     }
   },
 })
+
+function getSelector(
+  context: Readonly<TSESLint.RuleContext<MessageIds, Options>>,
+  { strict }: Options[number],
+) {
+  if (strict) {
+    return `${createEffectExpression} CallExpression > Identifier[name='withLatestFrom']` as const
+  }
+
+  const actionsName = findNgRxEffectActionsName(context)
+  return actionsName
+    ? (`${createEffectExpression} ${storeExpression(
+        actionsName,
+      )} > CallExpression[arguments.length=1] > Identifier[name='${withLatestFromKeyword}']` as const)
+    : null
```

Comment:
1. Do you see any problem in getting the `Actions` to report only in `Effects` (`strict: false`) that deal with actions? It was the cleanest way I could think of.
2. Ehh, maybe we should rename this selector in another PR. Actually `storeExpression` represents something named `XYZ` that is injected in `constructor`, not necessarily `store: Store` or `<accessibility> <readonly> store: Store`. The same applies to all selectors who derive from this, i.e. `storeExpressionCallable`.

Reply:
1. 👍
2. also a 👍, naming things is hard 😅